src/sbd-common.c

```diff
@@ -38,6 +38,7 @@
 unsigned long	timeout_watchdog 	= SBD_WATCHDOG_TIMEOUT_DEFAULT;
 int		timeout_msgwait		= 2 * SBD_WATCHDOG_TIMEOUT_DEFAULT;
 unsigned long	timeout_watchdog_warn 	= calculate_timeout_watchdog_warn(SBD_WATCHDOG_TIMEOUT_DEFAULT);
+bool		timeout_watchdog_warn_is_default = true;
```

Comment:
As the default is not a fixed value 'do_calculate_timeout_watchdog_warn' would sound more intuitive for me.
We could even think of moving it into the calculate_timeout_watchdog_warn macro to make the code more compact.
If the macro gets too large we can make it a function in sbd-common.c as well.

Reply:
The global `timeout_watchdog_warn` has to be initialized to some value anyway. I'd prefer keeping the macro and making the initial value right and clear at the beginning.

Reply:
I'm fine with initialization at the beginning for documentary reasons (and maybe dumb code analysis tools) - although it wouldn't have to be done before the first use.
Wasn't talking about moving the initialization to the macro or function.
I was thinking rather about the logic if it should be calculated from timeout_watchdog or taken from an explicitly set value.
As not to access a global variable inside the macro it could be passed as 2nd parameter.

Reply:
Could you show me a pieced of code how you'd like to do that?

Reply:
Thought of moving the always same if clause into the macro:
```
#define calculate_timeout_watchdog_warn(timeout, calculate) \
	(calculate ? \
	(timeout < 5 ? 2 : \
	(timeout < (ULONG_MAX / 3) ? \
	(((unsigned long) timeout) * 3 / 5) : (((unsigned long) timeout) / 5 * 3))) : \
	timeout)
```
But as there is just a single use of it left it is quite questionable.

Reply:
Hrm, indeed. And it'd make it less readable, and it'd be confusing to reference a macro so-called `calculate_*` just to calculate nothing ...

Reply:
> Thought of moving the always same if clause into the macro:
> 
> ```
> #define calculate_timeout_watchdog_warn(timeout, calculate) \
> 	(calculate ? \
> 	(timeout < 5 ? 2 : \
> 	(timeout < (ULONG_MAX / 3) ? \
> 	(((unsigned long) timeout) * 3 / 5) : (((unsigned long) timeout) / 5 * 3))) : \
> 	timeout)
> ```
BTW, `timeout` as the fallback doesn't seem correct.

Reply:
True but as said not really interesting anymore as we'd just be using it once atm.